### PR TITLE
deployment script: build, deploy, stack name, region, changeset

### PR DESCRIPTION
**Pull request by ritchieritch42**
"deploy": "sam build && sam deploy --stack-name im-sam --region us-east-1 --confirm-changeset"

### CODE REVIEW

**Comment by Andrewdt97** [comment]
1. Feel free to merge the other PR.
2. Feel free to merge this one too, but we still need a scrip that deploys the web app into the bucket. Lmk if this doesn't make sense.

File: package.json [hunk 1]
```diff
@@ -7,15 +7,15 @@
     "test": "test",
     "authlocalinvoke": "sam local invoke IPAuthorizerLambda --event ip-authorizer/event.json --env-vars ip-authorizer/env.json",
     "backlocalinvoke": "sam local invoke BackendLambda --event backend-lambda/event.json --env-vars backend-lambda/env.json",
-    "apilocalinvoke": "sam local start-api --env-vars backend-lambda/env.json"
+    "apilocalinvoke": "sam local start-api --env-vars backend-lambda/env.json",
+    "deploy": "sam build && sam deploy --stack-name im-sam --region us-east-1 --resolve-s3 --confirm-changeset"
```
Comment (Andrewdt97) [inline package.json]: Nope not it. This is kinda confusing but this is the bucket that CF uses to store it's state. We definitely do not want it to be our web app bucket and we don't really care so feel free to delete this arg. The commands we need have nothing to do with CF/SAM so the approach of finding new args for this command are not what we want. See the section titled: Bonus: Deploying with AWS CLI in this tutorial: https://wolovim.medium.com/deploying-create-react-app-to-s3-or-cloudfront-48dae4ce0af

Comment (ritchieritch42) [reply package.json]: Here's the CloudFront domain for the react app:

https://d3j9s5m87dlxez.cloudfront.net

File: react-cors-spa/yarn.lock [hunk 2]
```diff
@@ -2613,6 +2613,15 @@ axe-core@^4.4.3:
   resolved "https://registry.npmjs.org/axe-core/-/axe-core-4.5.1.tgz"
   integrity sha512-1exVbW0X1O/HSr/WMwnaweyqcWOgZgLiVxdLG34pvSQk4NlYQr9OUy0JLwuhFfuVNQzzqgH57eYzkFBCb3bIsQ==
 
+axios@^1.7.5:
```
Comment (Andrewdt97) [inline react-cors-spa/yarn.lock]: Why do we have a yarn lock file?

File: package.json [hunk 3]
```diff
@@ -7,15 +7,15 @@
     "test": "test",
     "authlocalinvoke": "sam local invoke IPAuthorizerLambda --event ip-authorizer/event.json --env-vars ip-authorizer/env.json",
     "backlocalinvoke": "sam local invoke BackendLambda --event backend-lambda/event.json --env-vars backend-lambda/env.json",
-    "apilocalinvoke": "sam local start-api --env-vars backend-lambda/env.json"
+    "apilocalinvoke": "sam local start-api --env-vars backend-lambda/env.json",
+    "deploysam": "sam build && sam deploy --stack-name im-sam --resolve-s3 --region us-east-1 --confirm-changeset",
+    "deployapp": "cd react-cors-spa && npm run build && aws s3 sync build/ s3://im.react.app --delete"
```
Comment (Andrewdt97) [inline package.json:12]: Update this to clear out the bucket before we stick new files in to make sure we don't get artifacts laying around.

File: react-cors-spa/yarn.lock (hunk 2 above)
Comment (ritchieritch42) [reply react-cors-spa/yarn.lock]: It's the base package installer/manager for the react-cors-spa repo, forgot to remove this!

File: package.json (hunk 3 above)
Comment (ritchieritch42) [reply package.json:12]: for deploysam?

deployapp will delete any resources that don't match the local directories/files that are being pushed

Comment (Andrewdt97) [reply package.json:12]: Oh sick my b!

**Review by Andrewdt97 — APPROVED** [review]

**Pull request merged**